src/geometry.F90

```diff
@@ -956,9 +958,13 @@ subroutine handle_lost_particle(p, message)
 !$omp atomic
     n_lost_particles = n_lost_particles + 1
 
+    ! Count the total number of simulated particles
+    tot_n_particles = n_batches * gen_per_batch * n_particles
```

Comment:
To be exact, you can use the `work` global variable which reflects the actual number of particles for a given process (accounting for cases where the number of particles is not evenly divisible by the number of processes).